diff --git a/bindings/python/tfel/CMakeLists.txt b/bindings/python/tfel/CMakeLists.txt
index bf623156b..68caf5182 100644
--- a/bindings/python/tfel/CMakeLists.txt
+++ b/bindings/python/tfel/CMakeLists.txt
@@ -21,7 +21,8 @@ if(TFEL_NUMPY_SUPPORT)
       ${CONAN_LIBS} ${PYTHON_LIBRARIES})
   else(TFEL_USES_CONAN)
     target_link_libraries(TFELNumpySupport
-      PUBLIC ${Boost_NUMPY_LIBRARY} ${Boost_PYTHON_LIBRARY} ${PYTHON_LIBRARIES})
+      PUBLIC ${Boost_NUMPY_LIBRARY} ${Boost_PYTHON_LIBRARY}
+      ${TFEL_PYTHON_MODULES_PRIVATE_LINK_LIBRARIES})
   endif(TFEL_USES_CONAN)
 endif(TFEL_NUMPY_SUPPORT)
 
